API.md

```diff
@@ -0,0 +1,66 @@
+
+### `void begin(const uint8_t* resetValues = NULL)`
+
+Initializes the given interface and prepares the board for communication. The argument `resetValues` sets the values of power switches of all devices by default as 8-bit. One bit for each power switch: `1` is "on" and `0` is "off". If the `resetValues` argument is adjusted, then when the `reset()` method is called, the power switches are set by the `resetValues`. If the `resetValues` argument is omitted, the power switches are set to "off" state.
+
+Important: call this method at least once, in your `setup()`.
+
+### `void digitalPinWrite(uint8_t pin, bool value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") of one power switch. Function arguments:
+
+- `pin`: the ordinal number of the power switch on the board. It ranges from `0` to `7`. Use an optional value `ALL` to select and set the state of all power switches simultaneously.
+- `value`: selects the state of the power switch. Values "on" and "off" may be put in different data types: `HIGH` or `LOW` / `true` or `false` / `1` or `0`.
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. Use an optional value `ALL` to select and set the state of all power switches simultaneously. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `void digitalPortWrite(uint8_t value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") for all power switches. Function arguments:
+
+- `value`: 8-bit value for all 8 power switches. One bit for each power switch: `1` is "on" and `0` is "off".
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. Use an optional value `ALL` to select and set the state of all devices. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `bool digitalPinRead(uint8_t pin, uint8_t device = 0)`
+
+Reads the state ("on" or "off") of a single power switch of the device. Returns `true` or `false`. Function arguments:
+
+- `pin`: the ordinal number power switch on the device. It ranges from `0` to `7`.
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `uint8_t digitalPortRead(uint8_t device = 0)`
+
+Reads state (turn-on or turn-off) of all power switches of the device. Returns the 8-bit value of all 8 power switches. One bit for each power switch: `1` is "on" and `0` is "off". Function argument `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last number device. If the `device` argument is omitted, it controls only one Octofet board.
```

Comment:
This creates an illusion of real reading. However, it just returns the internal state. I think this might be useful but should be named otherwise to avoid confusion. Like `getSwitchState` or something

Reply:
I suggest these names of methods:

```cpp
bool getSwitchState(uint8_t switch, uint8_t device = 0);
uint8_t getSwitchStateDevice(uint8_t device = 0);
```